robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowPendingIntent.java

```diff
@@ -134,37 +186,30 @@ public int getFlags() {
     return flags;
   }
 
+  public boolean isCanceled() {
+    return canceled;
+  }
+
+  @Implementation
+  public String getTargetPackage() {
+    return getCreatorPackage();
+  }
+
+  @Implementation(minSdk = 17)
+  public String getCreatorPackage() {
+    return (creatorPackage == null)
+        ? savedContext.getPackageName()
+        : creatorPackage;
+  }
+
+  public void setCreatorPackage(String creatorPackage) {
+    this.creatorPackage = creatorPackage;
+  }
+
   @Override
   @Implementation
   public boolean equals(Object o) {
-    if (this == o) return true;
```

Comment:
This is very frustrating, all tests pass locally, but not on Travis. In particular Travis is somehow managing to fail a simple reference equality check.

    org.robolectric.shadows.ShadowPendingIntentTest > send_callOnFinished FAILED
    Argument(s) are different! Wanted:
    onFinished.onSendFinished(
        PendingIntent{368f99d3: 555b2041},
        <Capturing argument>,
        101,
        <any string>,
        <any android.os.Bundle>
    );
    -> at org.robolectric.shadows.ShadowPendingIntentTest.send_callOnFinished(ShadowPendingIntentTest.java:155)
    Actual invocation has different arguments:
    onFinished.onSendFinished(
        PendingIntent{368f99d3: 555b2041},
        Intent {  },
        101,
        null,
        null
    );
    -> at org.robolectric.shadows.ShadowPendingIntent$OnFinishedRunnable.run(ShadowPendingIntent.java:362)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at org.robolectric.shadows.ShadowPendingIntentTest.send_callOnFinished(ShadowPendingIntentTest.java:155)

By looking at the dump PendingIntent{368f99d3: 555b2041} it's obvious that these are EXACTLY the same object, but mockito is somehow not considering them equal(). I would suspect this is a mockito bug given that this test passes when I run a copy of ShadowPendingIntentTest locally. Going to see if using same() instead work.